### ISSUE TITLE
Negative speed not working

### ISSUE THREAD

**Issue by riccardocescon** [rive-app/rive-flutter#302]
I have a custom controller that overrides appy like this:

```dart
@override
  void apply(RuntimeArtboard artboard, double elapsedSeconds) {
    if (instance == null || !instance!.keepGoing) {
      isActive = false;
    }
    instance!
      ..animation.apply(instance!.time, coreContext: artboard, mix: mix)
      ..advance(elapsedSeconds * speedMultiplier);
  }
```
thanks to #125 

When i call the River widget and set the controller speed to a negative number such as:
`speedMultiplier: -1`
the animation simply is freezed, same result of putting that value to 0.
with other values such as 0.5, 1, 2 and other positive numbers it works perfectly

I am trying to reverse-animating a One-Shot animation

### Device & Versions (please complete the following information)
- Device: Android Physical device
- OS: Android 13
- Flutter Version: 
```
Flutter 3.9.0-10.0.pre.6 • channel master • https://github.com/flutter/flutter.git
Framework • revision 53dfd23b04 (3 days ago) • 2023-03-17 01:59:19 -0400
Engine • revision d9e65441ea
Tools • Dart 3.0.0 (build 3.0.0-339.0.dev) • DevTools 2.22.2
```

**Comment by HayesGordon** [comment]
Hi @riccardocescon, the editor recently got a new feature to control the speed of animations directly from there, you can also reverse animations. See this tweet: https://twitter.com/rive_app/status/1635765458807984128

I encourage you to make use of state machines instead of mixing animations directly with code when possible.

But for your instance where you want to reverse a one shot animation, the reason you're not seeing anything is because a one shot animation does not loop. So when you try to reverse from the start of the animation it won't do anything because the timeline can't "subtract" the elapsed time back to the end of the timeline (if that makes sense).

TLDR; what you'll need to do is first set the One Shot animation's timeline to the end, and then you can reverse it. Here is some sample code:

```dart
class ReverseController extends SimpleAnimation {
  final double speedMultiplier;

  ReverseController(
    String animationName, {
    double mix = 1,
    this.speedMultiplier = -1,
  }) : super(animationName, mix: mix);

  @override
  bool init(RuntimeArtboard artboard) {
    final success = super.init(artboard);
    instance!.advance(instance!.animation.endTime);
    instance!.animation.speed = speedMultiplier;
    return success;
  }
}
```

**Comment by riccardocescon** [comment]
Thank you!
I switched to a State Machine and eveything works, i am move through animations by setting inputs values just like in the example folder.
I am having issues by reversing the animations now.
There is no _controller.advance(value) for that, neither a _controller or animation.speed parameter.
Have i missed s reverse-example on the example folder?
